### ISSUE TITLE
Request TV show is only adding with Future episodes monitored

### ISSUE THREAD

**Issue by Flippage** [tidusjar/PlexRequests.Net#587]
#### Plex Requests.Net Version:
1.9.4

#### Operating System:
Windows 10 - 64bit

#### Mono Version:
???

#### Applicable Logs (from `/logs/` directory or the Admin page):

```
[2016-10-12.zip](https://github.com/tidusjar/PlexRequests.Net/files/524031/2016-10-12.zip)
```

#### Reproduction Steps:
Doesn't seem to be any specific steps. Just request a show and regardless of the settings it adds it with only Future episodes being monitored. Doesn't monitor selected seasons or attempt to download episodes.

Doesn't make a difference whether I make the request as an admin or just as a user. Same effect both ways.

**Comment by Collisionc** [comment]
If the show has multiple seasons, it seems to add everything from previous seasons, but the current season it only adds future seasons.

**Comment by Flippage** [comment]
I'm actually not getting that myself @Collisionc - Mine won't automatically download anything. Even if I pick "All Seasons", it only monitors the Future episodes of the latest season (not even the entire latest season).

So weird, because CouchPotato I just set up, and that worked perfectly with a test movie, and in my experience CouchPotato has got nothing on Sonarr. Would have expected it the other way around haha.

**Comment by whitecollar17** [comment]
@tidusjar you broke something with the last update. If you request "all seasons" only the Future episodes got a monitoring status in sonarr.
If you have just one season. Nothing happen. Only the tv show is added to sonarr.
Is there a possibility to make a hotfix?

**Comment by tidusjar** [comment]
I am not 100% sure what is going on here, 
Yes this is caused by the update because we are using the API slightly different (Requesting the whole series rather than individual episodes).

I have raised a issue with Sonarr, since I don't think the API is working as expected. For now I have coded a work around.

I think the main issue was when requesting all series, it wasn't actually monitoring anything, I have placed a workaround for that, but I can't figure out why it is always monitoring the latest series but not any of the episodes, when I didn't ask it to do any of that.

I'll keep you guys updated.

If you want to use the workaround you are welcome to try the dev build

https://ci.appveyor.com/project/tidusjar/requestplex/build/artifacts

**Comment by Flippage** [comment]
Awesome thanks. Just as a note as well, I have tried selecting individual seasons / episodes, and the same issue occurs.

Thanks for all your work @tidusjar - it's a great piece of software either way :+1:

**Comment by whitecollar17** [comment]
@tidusjar  i run your dev build.
If you request something, i got a  "something went wrong".
After that i got under log error but the serie is added on ssonarr


[Newtonsoft.Json.JsonSerializationException: Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'PlexRequests.Api.Models.Sonarr.SonarrAddSeries' because the type requires a JSON object (e.g. {"name":"value"}) to deserialize correctly. To fix this error either change the JSON to a JSON object (e.g. {"name":"value"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array. Path '', line 1, position 1. at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureArrayContract(JsonReader reader, Type objectType, JsonContract contract) at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id) at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue) at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent) at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType) at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings) at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings) at PlexRequests.Api.ApiRequest.ExecuteJson[T](IRestRequest request, Uri baseUri) in C:\projects\requestplex\PlexRequests.Api\ApiRequest.cs:line 124 at PlexRequests.Api.SonarrApi.<>c__DisplayClass7_1.b__2() in C:\projects\requestplex\PlexRequests.Api\SonarrApi.cs:line 124 at Polly.Policy.<>c__DisplayClass8_0`1.b__0() at Polly.RetrySyntax.<>c__DisplayClass14_0.b__1() at Polly.Retry.RetryEngine.Implementation[TResult](Func`1 action, IEnumerable`1 shouldRetryExceptionPredicates, IEnumerable`1 shouldRetryResultPredicates, Func`1 policyStateFactory) at Polly.RetrySyntax.<>c__DisplayClass14_1.b__0(Action action, Context context) at Polly.Policy.Execute[TResult](Func`1 action, Context context) at PlexRequests.Api.SonarrApi.AddSeries(Int32 tvdbId, String title, Int32 qualityId, Boolean seasonFolders, String rootPath, Int32 seasonCount, Int32[] seasons, String apiKey, Uri baseUrl, Boolean monitor, Boolean searchForMissingEpisodes) in C:\projects\requestplex\PlexRequests.Api\SonarrApi.cs:line 124](url)

![image](https://cloud.githubusercontent.com/assets/2944704/19492061/8ac4c66a-9575-11e6-9329-7fd6da675a4b.png)

**Comment by tidusjar** [comment]
@whitecollar17 do you want to try a new build?

If you get the same error can you put it on Trace and then get the error?
https://ci.appveyor.com/project/tidusjar/requestplex/build/artifacts

**Comment by whitecollar17** [comment]
I can test te code on sunday.  I'm away from home this weekend.  @tidusjar

**Comment by tidusjar** [comment]
Thats cool @whitecollar17 !

**Comment by whitecollar17** [comment]
@tidusjar i run the new build now, but got the same error after waiting some time. Got a "something went wrong error".

Log error set to trace, this is what i go t in the logs of plexrequest.net:



[logdump.txt](https://github.com/tidusjar/PlexRequests.Net/files/545860/logdump.txt)

**Comment by tidusjar** [comment]
Just to clarify, what did you press before something went wrong?

**Comment by whitecollar17** [comment]
I press Request all seasons. After 10 seconds this popup shows up with something went wrong

**Comment by Collisionc** [comment]
Yeah, I'm seeing the same thing. I had to disable tv shows temporarily as it crashes the whole thing.

**Comment by tidusjar** [comment]
I'm still waiting for a reply for the Sonarr guys about this...

**Comment by Collisionc** [comment]
No rush, just thought I'd mention that I see the same thing as well.

**Comment by whitecollar17** [comment]
suggesties off me: rollback the code and wait for a fix on the side from sonarr

**Comment by tidusjar** [comment]
Already done @whitecollar17 !

**Comment by whitecollar17** [comment]
@tidusjar i tested the rollback, looking good. No problems here on the last dev build!

**Comment by tidusjar** [comment]
Excellent!

**Comment by whitecollar17** [comment]
Fixed in the release of 1.9.4
